### PR TITLE
Add a simple model for Tags

### PR DESCRIPTION
**Pull request by okkays**
https://github.com/usdoj-crt/crt-portal-management/issues/1656

## What does this change?

- 🌎 We're adding tags to reports
- ⛔ We have no database backing for those, yet
- ✅ This commit adds a simple DB object and association on reports.
- 🔮 This is part of a larger tags MVP, future commits will use these models in the UI.

## Screenshots (for front-end PR):

Admin panel fields

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/models.py [hunk 1]
```diff
@@ -255,6 +255,23 @@ def __str__(self):
...
 
+class Tag(models.Model):
+
+    class Meta:
+        permissions = (
+            ("assign_tag", "Can assign tags to reports"),
+        )
+
+    name = models.CharField(max_length=255, null=False, blank=False, help_text="The name of the tag that will be shown throughout the app.")
+    section = models.TextField(choices=SECTION_CHOICES, null=True, blank=True, default=None, help_text="The section to which this tag applies. If set, this tag will only be available to reports in this section.")
```
Comment (kewitham) [inline crt_portal/cts_forms/models.py:266]: Do we want to allow tags to be used by multiple sections?

Comment (okkays) [reply crt_portal/cts_forms/models.py:266]: I think so, yeah. But in that case, I anticipated making the tag available to all sections (by setting section to `null`)

If we want to allow a tag to a few sections (but not all), it'd mean that we need a many-to-many relationship, which would mean we'd be executing a join every time we queried for tags based on section.

Probably not the worst, but it's additional complexity we can avoid and I'm not sure the business case of sharing tags between two sections is worth it.

Comment (okkays) [reply crt_portal/cts_forms/models.py:266]: (@blaughman curious to hear your take)

Comment (blaughman) [reply crt_portal/cts_forms/models.py:266]: @okkays, Amanda, @ilodidoj  and I discussed this during today's standup.  I think having the section acronym associated with the label would be helpful. As we saw with secondary review, universal labels may have different meanings with different sections. I don't want to recreate that same experience. I'm attaching a file SPL shared with me detailing their acronyms for labeling reports and the associated definitions.  Hope this helps and I hope all is going well for you. 
[SPL ISSUES - UPDATED.pdf](https://github.com/usdoj-crt/crt-portal/files/13504289/SPL.ISSUES.-.UPDATED.pdf)

**Review by kewitham — APPROVED** [review]

**Pull request merged**